[QT] fix OSX dock icon window reopening

jonasschnelli opened the pull request:
fixes #5878.
Tested on OSX 10.10.

Needs testing on OSX 10.7 till 10.9.

jonasschnelli committed:
[QT] fix OSX dock icon window reopening

fixes #5878

jonasschnelli commented:
Binaries to test: https://builds.jonasschnelli.ch/pulls/5880/

jonasschnelli commented:
Tested https://builds.jonasschnelli.ch/pulls/5880/bitcoin-0.10.99-osx-unsigned.dmg on 10.7 and 10.9.
Works fine there.
Assume 10.8 will also work.

laanwj commented on src/qt/macdockiconhandler.mm:
```diff
@@ -11,52 +11,46 @@
-@end
-
-@implementation DockIconClickEventHandler
+static MacDockIconHandler *s_instance = NULL;
```
Does s_instance ever get freed?

jonasschnelli replied:
No. It's a singleton pattern and i assume the OS will care about the cleanup.
But i could try to proactive free it within the destructor of `~MacDockIconHandler`. But this might be more or less the same cleanup as the os does.

laanwj replied:
I'd prefer if it was freed. Otherwise it will show up as a memory leak. Would be even better to do it without introducing a singleton, but I suppose that isn't possible.

fanquake commented:
Tested ACK

laanwj commented:
@fanquake on what version of OSX?

fanquake commented:
@laanwj 10.9 & 10.10

jonasschnelli committed:
[QT] some mac specifiy cleanup (memory handling, unnecessary code)

jonasschnelli commented:
Updated with @laanwj nits (mem cleanup).
Also removed unused code.
@fanquake mind retesting.
Binaries: https://builds.jonasschnelli.ch/pulls/5880/

jonasschnelli commented:
Retested on 10.7 and 10.9. Works fine there.

fanquake commented:
ReACK Checked 10.9 & 10.10

laanwj commented:
Cherry-picked to 0.10 as 81145a6 and 2c08406
